src/pages/programDisaggregations/form/DissaggregationCategories.tsx

```diff
@@ -0,0 +1,460 @@
+            return {
+                ...ccToCat,
+                [cCombo.id]: cCombo.categories.map((category) => category.id),
+            }
+        }, {} as Record<string, string[]>)
+
+    const categoryObject = categoryCombosData?.categoryCombos?.reduce(
+        (allCategories, catCombo) => {
+            catCombo.categories.forEach((category) => {
+                if (!allCategories[category.id]) {
+                    allCategories[category.id] = category
+                }
+            })
+            return allCategories
+        },
+        {} as CategoryObject
+    )
+    const categoriesArray = Object.values(categoryObject || {}).sort((a, b) =>
+        a?.displayName?.localeCompare(b?.displayName)
+    )
+    const categoryCombosArray =
+        categoryCombosData?.categoryCombos?.map((catCombo) => ({
+            id: catCombo.id,
+            displayName: catCombo.displayName,
+        })) || []
+
+    return {
+        categoryCombosToCategoriesObject,
+        categoryObject,
+        categoriesArray,
+        categoryCombosArray,
+    }
+}
+
+type optionsMappingArray = Record<string, { optionId: string; filter: string }>
+const getEmptyOptionValues = (categoryOptions: CategoryOption[]) => {
+    return categoryOptions?.reduce(
+        (optionObject: optionsMappingArray, option: CategoryOption) => {
+            optionObject[option.id] = {
+                optionId: option.id,
+                filter: '',
+            }
+            return optionObject
+        },
+        {} as optionsMappingArray
+    )
+}
+
+type ProgramIndicatorValue = Record<
+    string,
+    {
+        categoryCombo: {
+            categories: { id: string }[]
+        }
+    }
+>
+
+// const getCategoriesFromProgramIndicatorSelections = (
+//     programIndicatorsValue: ProgramIndicatorValue,
+//     categoriesWithMappings: string[]
+// ) => {
+//     const programCategories = Object.values(programIndicatorsValue)
+//         ?.map((piValue) => {
+//             return piValue?.categoryCombo?.categories
+//         })
+//         ?.flat()
+//         ?.map((category) => category?.id)
+//     return [...new Set(programCategories)].filter(
+//         (id) => !categoriesWithMappings.includes(id)
+//     )
+// }
+
+const useProgramIndicatorSelectionCategories = (
+    categoriesWithMappings: string[]
+) => {
+    const programIndicatorsValue: ProgramIndicatorValue = useField(
+        'programIndicatorMappings'
+    )?.input?.value
+    const programCategories = Object.values(programIndicatorsValue)
+        ?.map((piValue) => {
+            return piValue?.categoryCombo?.categories
+        })
+        ?.flat()
+        ?.map((category) => category?.id)
+    return [...new Set(programCategories)].filter(
+        (id) => !categoriesWithMappings.includes(id)
+    )
+}
+
+export const DisaggregationCategories = () => {
+    const formApi = useForm()
+
+    // this is to have a stable-ish reference instead of taking from formState (rethink)
```

Comment:
If we want the more stable reference here, it would be better to get the populated array of initial category mapping ids from the parent <Edit> component rather than the approach here of using a useEffect, but I didn't want to touch the Edit component to prevent divergence issues.